core/src/main/java/org/frankframework/mongodb/StrictJsonDocumentWriter.java

```diff
@@ -104,7 +103,8 @@ public void writeStartObject() {
 			throw new BsonInvalidOperationException("Invalid state " + state);
 		}
 		try {
-			INodeBuilder nodeBuilder = context.contextType == JsonContextType.ARRAY ? ((IArrayBuilder)stack.peek()).addElement() : (INodeBuilder)stack.peek();
+			INodeBuilder nodeBuilder = context.contextType == JsonContextType.ARRAY ? ((IArrayBuilder)stack.peek()).addElement() : (INodeBuilder)stack.pop();
+			stack.push(nodeBuilder);
```

Comment:
This line is the real fix. The `nodeBuilder` itself was not added while starting an array object.